### PR TITLE
Button Bar Pattern

### PR DESCRIPTION
**Pull request by lilyvc**
Button bar pattern:

This PR 
- adds the button bar page
- adds storybook examples of select button bar implementations

### CODE REVIEW

**Review by origami-z — COMMENTED** [review]
A few examples are wrapped with `width: "40vw"`. Are we worried about people just copy/paste code and completely ditch responsiveness (where our recommendation is)? e.g. "Single Step Form" example doesn't do stack. 

In my mind every single example should have responsiveness built in. There's storybook responsive container toggle which this could benefit, which i don't know how much of these can be reused or whether it can be set as default for button bar stories.

File: site/docs/patterns/button-bar.mdx [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+Button Bar is a pattern that displays possible actions for users to take when they reach the end of a task. You can use it to provide consistent design when implementing actionable interactions within pages or dialogs across your applications.
+
+  <Diagram
+    src="/img/patterns/button-bar/with-secondary.png"
+    alt="Example button bar, with CTA to the left then primary, on the right side there are two secondary buttons"
+    contentPosition="top"
+    border
+  />
+
+We provide design guidance of how you can use Salt Button and Layout components and Button Bar template to achieve the pattern. Implementation examples can also be seen in the storybook.
```
Comment (feliciackh) [inline site/docs/patterns/button-bar.mdx:22]: would it be beneficial to link to Salt components whenever they're mentioned in this pattern

Comment (lilyvc) [reply site/docs/patterns/button-bar.mdx:22]: yeah I think for the initial mention it makes sense. Although, Layout isn't something we can link to directly so would have to mention all the layouts used

File: site/src/layouts/DetailPattern/Resources.tsx [hunk 2]
```diff
@@ -12,6 +12,9 @@ const LinkWithLogo = ({ href, label }: { href: string; label: string }) => (
     {href.includes("github.com") && (
       <Image src="/img/github_logo.svg" alt="github logo" />
     )}
+    {href.includes("storybook") && (
+      <Image src="/img/storybook_logo.svg" alt="github logo" />
```
Comment (feliciackh) [inline site/src/layouts/DetailPattern/Resources.tsx:16]: alt="storybook logo"

File: site/docs/patterns/button-bar.mdx [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+The container and the type of scenario determine the position of actions (CTA, primary and secondary buttons).
+
+<Diagrams>
+  <Diagram
+    src="/img/patterns/button-bar/default-order.png"
+    alt="Example showing default button bar order, submit CTA on the left and cancel primary to the right of submit"
+    contentPosition="top"
+    border
+  >
+    #### Default/Single-step form
```
Comment (joshwooding) [inline site/docs/patterns/button-bar.mdx:106]:
```suggestion
    ### Default/Single-step form
```

File: site/docs/patterns/button-bar.mdx [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+    - Groups core actions on the left.
+    - Follows an F-pattern reading model, allowing users to quickly scan down.
+  </Diagram>
+  <Diagram
+    src="/img/patterns/button-bar/multi-step-form.png"
+    alt="Example showing form with button bar underneath, the core actions are grouped on the right."
+    contentPosition="top"
+    border
+  >
+    #### Multi-step form
```
Comment (joshwooding) [inline site/docs/patterns/button-bar.mdx:116]:
```suggestion
    ### Multi-step form
```

File: site/docs/patterns/button-bar.mdx [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+    - Always displayes core actions on the right.
+    - Follows a Z-pattern reading model, where actions are the last focal point in the dialog.
+  </Diagram>
+  <Diagram
+    src="/img/patterns/button-bar/small-viewport.png"
+    alt="Example showing a form with two buttons that fill the width"
+    contentPosition="bottom"
+    border
+  >
+    #### Small viewport (below the xs breakpoint)
```
Comment (joshwooding) [inline site/docs/patterns/button-bar.mdx:136]:
```suggestion
    ### Small viewport (below the xs breakpoint)
```

**Comment by lilyvc** [comment]
> A few examples are wrapped with `width: "40vw"`. Are we worried about people just copy/paste code and completely ditch responsiveness (where our recommendation is)? e.g. "Single Step Form" example doesn't do stack.
> 
> In my mind every single example should have responsiveness built in (unless we have pattern doc specifically says when not to). There's storybook responsive container toggle which this could benefit, which i don't know how much of these can be reused or whether it can be set as default for button bar stories.
> 
> storybook responsiveness toggle

the responsive container doesn't seem to act responsive, so I'm not sure it's an immediate solution.
the button bar won't be full width unless it is wrapped in something with some value of vw, I imagine if people copy the code they will just take the button bar part and put it inside their own container

**Review by ImmanuelBaskaran — APPROVED** [review]

**Review by feliciackh — APPROVED** [review]

**Pull request merged**